test_engine.py

```diff
@@ -0,0 +1,372 @@
+                            min_zoom = 0.8, 
+                            max_zoom = 1.2, 
+                            allow_missing_keys = True
+                            ),
+                        RandFlipd(
+                            Keys.all(),
+                            prob = 0.5, 
+                            spatial_axis = 1, 
+                            allow_missing_keys = True
+                            ),
+                        RandRotated(
+                            Keys.all(),
+                            range_x = 1.5, 
+                            range_y = 0, 
+                            range_z = 0, 
+                            prob = 0.5, 
+                            allow_missing_keys = True
+                            ),
+                        RandAdjustContrastd(Keys.IMAGE, prob = 0.25),
+                        NormalizeIntensityD(Keys.IMAGE, channel_wise = True),
+                        ForegroundMaskD(Keys.LABEL, threshold = 0.5, invert = True),
+                        ToTensorD(Keys.all(), allow_missing_keys = True),
+                    ]
+                )
+    
+    engine.test_transform = Compose(
+                [
+                    LoadImageD(Keys.all(), allow_missing_keys = True),
+                    EnsureChannelFirstD(Keys.all(), allow_missing_keys = True),
+                    ResizeD(
+                        Keys.all(),
+                        spatial_size=[256, 256,128], 
+                        mode=("trilinear", "nearest", "nearest"), 
+                        allow_missing_keys = True,
+                    ),
+                    NormalizeIntensityD(Keys.IMAGE, channel_wise = True),
+                    ForegroundMaskD(
+                        Keys.LABEL,
+                        threshold = 0.5, 
+                        invert = True, 
+                        allow_missing_keys = True
+                    ),
+                    ToTensorD(Keys.all(), allow_missing_keys = True),
+                ]
+            )
+    engine.postprocessing_transforms = Compose([
+    ActivationsD(Keys.PRED,sigmoid = True),
+    AsDiscreteD(Keys.PRED,threshold = 0.5)
+    
+])
+#########################################################################################################################
+def test_load_data(engine):
+    """
+    Description: Tests the load_data method of the engine.
+
+    Parameters: engine(object) : An instance of the Engine class.
+
+    Returns: None
+    """
+
+    get_transforms(engine)
+    engine.load_data()
+
+    assert len(engine.train_dataloader) > 0
+    assert len(engine.test_dataloader) > 0
+    assert isinstance(engine.train_dataloader, MonaiLoader)
+    assert isinstance(engine.val_dataloader, MonaiLoader)
+    assert isinstance(engine.test_dataloader, MonaiLoader)
+
+
+#########################################################################################################################
+def test_save_checkpoint(engine,ckpt_path='Checkpoint'):
```

Comment:
Maybe you can test the loading function first, then make sure that saving and reloading a checkpoint yield the same weights.

Reply:
done